githttp.go

```diff
@@ -10,6 +10,14 @@ import (
 	"strings"
 )
 
+type ResolveInfo struct {
+	Path    string
+	Request *http.Request
+}
```

Comment:
Are you using a struct here instead of making `PathResolver` accept those 2 parameters because you expect there may be changes, and you want to be able to do that without breaking the API?